core/engine/engine_test.go

```diff
@@ -67,8 +67,8 @@ func mkTestHost() host.Host {
 // testing purposes. A more complex callback could read
 // feom the CID index and return the list of CIDs.
 func simpleCb(cids []cid.Cid) core.CidCallback {
-	return func(c cid.Cid) []cid.Cid {
-		return cids
+	return func(c cid.Cid) ([]cid.Cid, error) {
```

Comment:
eventually we'll want the `[]cid.Cid` to be instead perhaps a `chan cid.Cid` so we can stream the list as we need it?